Philips.CodeAnalysis.MaintainabilityAnalyzers/Readability/AvoidMultipleConditionsOnSameLineAnalyzer.cs

```diff
@@ -0,0 +1,127 @@
+			"Divide multiline conditions around line {0} such, that the logical operators are at the end " +
+			"of the line.";
+		private const string Description = "Divide long conditions in consistent way";
+		private const string Category = "Shoscar";
+
+		private static readonly DiagnosticDescriptor Rule =
+			new (
+				Helper.ToDiagnosticId(DiagnosticIds.AvoidMultipleConditionsOnSameLine),
+				Title,
+				Message,
+				Category,
+				DiagnosticSeverity.Warning,
+				isEnabledByDefault: true,
+				description: Description);
+
+		/// <summary>
+		/// <inheritdoc cref="DiagnosticAnalyzer.SupportedDiagnostics"/>
+		/// </summary>
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+			ImmutableArray.Create(Rule);
+
+		/// <summary>
+		/// <inheritdoc/>
+		/// </summary>
+		public override void Initialize(AnalysisContext context)
+		{
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+			context.EnableConcurrentExecution();
+			context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.LogicalAndExpression);
+			context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.LogicalOrExpression);
+		}
+
+		private void AnalyzeNode(SyntaxNodeAnalysisContext context)
+		{
+			GeneratedCodeDetector generatedCodeDetector = new();
+			if (generatedCodeDetector.IsGeneratedCode(context))
+			{
+				return;
+			}
+
+			var logicalNode = context.Node;
+			var location = logicalNode.GetLocation();
+			if (!IsMultiLine(location))
+			{
+				return;
+			}
+			
+			var childNodes = logicalNode.ChildNodes();
+			bool hasCombinedNode = childNodes.OfType<ParenthesizedExpressionSyntax>().Any();
+			if (hasCombinedNode)
+			{
+				return;
+			}
+
+			// Checked in parts:
+			// 1) first line should only have open parentheses.
+			// 2) if newline is on operator or open parentheses.
+			var tokenBefore = logicalNode.GetFirstToken().GetPreviousToken();
+			var check1Pass = ContainsEndOfLine(tokenBefore);
+			if (!check1Pass)
+			{
+				ReportDiagnostic(context, tokenBefore);
+			}
+
+			var violations = logicalNode.DescendantTokens()
+				.Where(ContainsEndOfLine)
+				.Where(IsIllegalLineBreakToken); // Check 2).
+			foreach (var violation in violations)
+			{
+				ReportDiagnostic(context, violation);
+			}
+		}
+
+		private void ReportDiagnostic(SyntaxNodeAnalysisContext context, SyntaxToken violation)
+		{
+			var loc = violation.GetLocation();
+			var lineNumber = loc.GetLineSpan().StartLinePosition.Line + 1;
+			context.ReportDiagnostic(Diagnostic.Create(Rule, loc, lineNumber));
+		}
+
+		private bool IsMultiLine(Location loc)
+		{
+			var lineSpan = loc.GetLineSpan();
+			var startLine = lineSpan.StartLinePosition.Line;
+			var endLine = lineSpan.EndLinePosition.Line;
+			return (startLine != endLine);
+		}
+
+		private bool ContainsEndOfLine(SyntaxToken token)
+		{
+			return token.TrailingTrivia.Any(SyntaxKind.EndOfLineTrivia);
+		}
+
+		private bool IsIllegalLineBreakToken(SyntaxToken token)
```

Comment:
This was a finding w/o a clear solution other than to create a long line (or distribute the !, which seems beyond the intent of this analyzer):
			if (!(StringHelper.StartsWith(messagesText, HL7Message.MessageStartDelimeter) &&
				StringHelper.EndsWith(messagesText, HL7Message.MessageEndDelimeter)))

Reply:
The only way I see this case can be supported, is to allow the `DotToken` as last token on a line.
But there is a significant downside in doing this, as it becomes unreadable if this is done for short expressions.

For example, this code would be valid than:
``` cs
if (a.Property.
  ToString()) 
{
  // blah
}
```

Open for suggestions here.